subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -71,6 +71,20 @@ enum {
 	ADV_NUM_FLAGS,
 };
 
+struct delayable_adv_ctx {
+	sys_snode_t node;
+	struct bt_mesh_adv *adv;
+	const struct bt_mesh_send_cb *cb;
+	void *cb_data;
+	bool retransmit;
+	uint32_t fired_time;
+};
+
+K_MEM_SLAB_DEFINE(delayable_advs_buf_pool,
```

Comment:
> Is any memory management required at all? There is only one sending packet at the same time

If I don't miss anything, acks can be send in parallel to regular PDUs in case an ack is lost by one role, but the other role started sending next PDU, e.g.:
https://github.com/zephyrproject-rtos/zephyr/blob/fe75bb950f45c1740225a21a2638a20080f4e967/subsys/bluetooth/mesh/pb_adv.c#L453-L461

I tend to think that every allocated adv buf requires only `sys_snode_t` entry which is consecutively attached to a single list. The `k_work_delayable` then pulls entries from this list one after another thus also adhering the order of the PDUs which you pointed out about in https://github.com/zephyrproject-rtos/zephyr/pull/65629/files#r1439302794.

Reply:
I have tried looking if I could remove this, but I decided that it was best to keep it. I have removed the slab allocator.
The reason is this way everything goes through the same functions, and I do not have to do anything on tside for sending the ack and the unacked messages. For both of them I would have had to keep the message, callback and user_data in seperate data structures. I also would have had to some how kept a record of what messages for tx.adv we have sent and not sent. As they have to be kept for retransmissions. I figured out that this was the easyest way of keeping a record on what need to be sent.